python/python/lance/dataset.py

```diff
@@ -823,6 +826,11 @@ def create_index(
         self._ds.create_index(column, index_type, name, metric, kwargs)
         return LanceDataset(self.uri)
 
+    def optimize_indices(self, **kwargs) -> LanceDataset:
+        """Optimize indices."""
+        self._ds.optimize_indices(kwargs)
+        return LanceDataset(self.uri)
```

Comment:
Could we keep this just behind `dataset.optimize`? trying to keep the table optimization methods grouped together.

Reply:
Sure thing. 

How about `dataset.optimize.indices()` ? As `dataset.optimize` is a verb already.